src/device/device-management.ts

```diff
@@ -33,29 +33,87 @@ export type AudioDeviceConstraints = {
   deviceId?: string;
 };
 
-export type VideoDeviceConstraints = {
-  deviceId?: ConstrainDOMString;
-  width?: ConstrainULong;
-  height?: ConstrainULong;
-  aspectRatio?: ConstrainDouble;
-  frameRate?: ConstrainDouble;
-  facingMode?: ConstrainDOMString;
+// Predefined encoder config mentioned similar to https://api-ref.agora.io/en/voice-sdk/web/4.x/globals.html#videoencoderconfigurationpreset
+export enum StaticVideoEncoderConfig {
+  Profile1080P = '1080P',
+}
+
+export type ConstrainLong = {
+  min?: number;
+  max?: number;
+  ideal?: number;
+  exact?: number;
+};
+
+// Predefined encoder config mentioned similar to https://api-ref.agora.io/en/voice-sdk/web/4.x/globals.html#videoencoderconfigurationpreset
+export type videoEncoderConfig = {
```

Comment:
This should start with a capital letter.

Reply:
done